packages/block-library/src/footnotes/format.js

```diff
@@ -34,23 +35,26 @@ export const format = {
 		'data-fn': 'data-fn',
 	},
 	contentEditable: false,
-	[ usesContextKey ]: [ 'postType' ],
+	[ usesContextKey ]: [ 'postType', 'postId' ],
 	edit: function Edit( {
 		value,
 		onChange,
 		isObjectActive,
-		context: { postType },
+		context: { postType, postId },
 	} ) {
 		const registry = useRegistry();
 		const {
 			getSelectedBlockClientId,
 			getBlockRootClientId,
 			getBlockName,
-			getBlocks,
+			getBlockParentsByBlockName,
 		} = useSelect( blockEditorStore );
 		const footnotesBlockType = useSelect( ( select ) =>
 			select( blocksStore ).getBlockType( name )
 		);
+		const [ blocks ] = useEntityBlockEditor( 'postType', postType, {
```

Comment:
`blocks` is only used in the onClick callback. Is there a selector that we could run there instead?

Reply:
Oh I was just writing the [comment](https://github.com/WordPress/gutenberg/pull/53003/files#r1275542391) below!

Looking into it now. Thanks!

Reply:
I need to look further, but I'm having trouble finding an alternative that gives me the rendered block list for a page in the site editor. It's definitely a limitation of my knowledge, I don't spend much time in core-data 😄 

The `blocks` property is available outside the click handler, but not within it for example

```js
// returns the right block list outside onClick but `undefined` within it
getEditedEntityRecord(
					'postType',
					postType,
					postId
				)?.blocks
```

I can see that we get the edited entity record with raw content, whose blocks could be parsed. Not sure if that's very performant either.

Reply:
> I can see that we get the edited entity record with raw content, whose blocks could be parsed. Not sure if that's very performant either

I think using `getEditedEntityRecord` gets us closer to the desired end state. From looking at `useEntityBlockEditor`, it currently does the parsing [here](https://github.com/WordPress/gutenberg/blob/dff466a315104ba36585f571f1d836643f789d95/packages/core-data/src/entity-provider.js#L180) if `blocks` isn't available, so performance-wise, if we copy/pasted (or followed) a similar approach in the `onClick` callback, it'd be no slower than using `useEntityBlockEditor`, I'd think.

> if we'd get the necessary clientId of the footnotes block.

It doesn't look like we need the clientId of the footnotes block, rather, we're just checking whether it exists. So I _think_ we should be fine using the parsed content 🤞

Reply:
> if we'd get the necessary clientId of the footnotes block.

Oh, wait, that does seem to be problem unfortunately! Sorry I missed that in my first pass, without that clientId, we don't have selection move down to the footnote block 🤔